src/core/server/saved_objects/service/lib/aggs_utils.test.ts

```diff
@@ -0,0 +1,249 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { validateGetSavedObjectsAggs } from './aggs_utils';
+import { mockMappings } from './filter_utils.test';
+
+describe('Filter Utils', () => {
+  describe('#validateGetSavedObjectsAggs', () => {
+    test('Validate a simple aggregations', () => {
+      expect(
+        validateGetSavedObjectsAggs(
+          ['foo'],
+          { aggName: { max: { field: 'foo.attributes.bytes' } } },
+          mockMappings
+        )
+      ).toEqual({
+        aggName: {
+          max: {
+            field: 'foo.bytes',
+          },
+        },
+      });
+    });
+
+    test('Validate a nested field in simple aggregations', () => {
+      expect(
+        validateGetSavedObjectsAggs(
+          ['alert'],
+          { aggName: { cardinality: { field: 'alert.attributes.actions.group' } } },
+          mockMappings
+        )
+      ).toEqual({
+        aggName: {
+          cardinality: {
+            field: 'alert.actions.group',
+          },
+        },
+      });
+    });
+
+    test('Validate a nested aggregations', () => {
```

Comment:
I know you fixed it with the 'manual' recursion validation, but can we still add a test with a 3 level deep aggregation?